test/gurka/test_time_dependent_restrictions.cc

```diff
@@ -0,0 +1,60 @@
+#include "gurka.h"
+#include "proto/api.pb.h"
+
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+class TimeDependentRestrictions : public testing::TestWithParam<int> {
+public:
+  static gurka::map map;
+
+  static void SetUpTestSuite() {
+    const std::string ascii_map = R"(
+          D----B----A
+          |    |
+          |    |
+          E----C----F
+     )";
+
+    const gurka::ways ways = {
+        {"AB", {{"highway", "motorway"}}}, {"BC", {{"highway", "motorway"}}},
+        {"BD", {{"highway", "motorway"}}}, {"CF", {{"highway", "motorway"}}},
+        {"DE", {{"highway", "motorway"}}}, {"EC", {{"highway", "motorway"}}},
+    };
+    const gurka::nodelayout layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+    const gurka::relations relations = {{{
+                                             {gurka::way_member, "AB", "from"},
+                                             {gurka::way_member, "BC", "to"},
+                                             {gurka::node_member, "B", "via"},
+                                         },
+                                         {{"type", "restriction"},
+                                          {"restriction", "no_left_turn"},
+                                          {"hour_on", "07:00"},
+                                          {"hour_off", "19:00"}}}};
+    map =
+        gurka::buildtiles(layout, ways, {}, relations, "test/data/gurka_time_dependent_restrictions",
+                          {{"mjolnir.timezone", VALHALLA_BUILD_DIR "test/data/tz.sqlite"}});
+  }
+};
+gurka::map TimeDependentRestrictions::map = {};
+
+// TODO: Does not work for date_time_type=2, returns additional AB edge.
```

Comment:
it's because of this https://github.com/valhalla/valhalla/issues/2940 ; try to put `start` and `end` locations on the edges
```
          D----B---1-A
          |    |
          |    |
          E----C---2-F
```

Reply:
@genadz since `date_time: 2` uses timedep-reverse, I wonder if the extra edge is being returned due to the bug @danpat found [here](https://github.com/valhalla/valhalla/pull/2987#issuecomment-824448011).

@CuriOusQuOkka https://github.com/valhalla/valhalla/pull/2987 fixed a bug in timedep-reverse, which might fix the problem you're seeing. If so, you can re-enable `date_time: 2`.

Reply:
Thanks @mandeepsandhu, it fixed the bug.

Reply:
> @genadz since date_time: 2 uses timedep-reverse, I wonder if the extra edge is being returned due to the bug @danpat found here.

cool! so, I should close this issue #2940